server/lib/notifications.js

```diff
@@ -215,9 +215,61 @@ const notifyUpdateSubscribers = async activity => {
   const emailOpts = { from: activity.data.fromEmail };
   const update = await models.Update.findByPk(activity.data.update.id);
   const allUsers = await update.getUsersToNotify();
-  return notifySubscribers(allUsers, activity, emailOpts);
+  const modifiedActivity = replaceVideosByImagePreviews(activity);
+  return notifySubscribers(allUsers, modifiedActivity, emailOpts);
 };
 
+function replaceVideosByImagePreviews(activity) {
+  const iframePreviewRegex = /<iframe\s[^>]*src="([^"]+)"[^>]*><\/iframe>/gi;
+  activity.data.update.html = activity.data.update.html
+    .replace(iframePreviewRegex, (_, href) => {
+      const { service, id } = parseServiceLink(href);
+      const imgSrc = constructPreviewImageURL(service, id);
+      if (imgSrc) {
+        return `<img src="${imgSrc}" alt="${service} content">`;
+      } else {
+        return '';
+      }
+    })
+    .replace(/<iframe\s[^>]><\/iframe>/gi, '');
+  return activity;
+}
+
+function constructPreviewImageURL(service, id) {
+  if (service === 'youtube') {
+    return `https://img.youtube.com/vi/${id}/0.jpg`;
```

Comment:
In the spirit of defensive programming, I think we should sanitize the `id` here. A YouTube id is `[a-zA-Z0-9_-]{11}`, if we don't get something like that we should return `null`.

Reply:
Thanks. I've changed that to sanitize using the regex you suggested. 👍🏽